app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,36 @@
+class QueueItemsController < ApplicationController
+  before_filter :require_user
+
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    video = Video.find(params[:video_id])
+    queue_video(video)
+    redirect_to my_queue_path
+  end
+
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    queue_item.destroy if queue_item.user == current_user
+    update_order(current_user.queue_items.order(:order))
```

Comment:
I don't see any reason to need to pass these in, since the `update_order` method is in the same scope, you can just access the items within the method.